lib/babe/build.go

```diff
@@ -113,7 +113,9 @@ func (b *BlockBuilder) buildBlock(parent *types.Header, slot Slot, rt runtime.In
 
 	// create new block header
 	number := big.NewInt(0).Add(parent.Number, big.NewInt(1))
-	header, err := types.NewHeader(parent.Hash(), common.Hash{}, common.Hash{}, number, types.NewDigest(preDigest))
+	digest := types.NewDigest()
+	_ = digest.Add(*preDigest)
```

Comment:
I noticed the error is often ignored for `digest.Add()`, is there a chance an error would occur?

Reply:
this error should be handled.  In the event the inputted param doesn't fulfil interface `VaryingDataTypeValue`, then it will error out.